python/hail/representation/genomeref.py

```diff
@@ -0,0 +1,200 @@
+                      y_contigs=oneof(strlike, listof(strlike)),
+                      mt_contigs=oneof(strlike, listof(strlike)),
+                      par=listof(Interval))
+    def __init__(self, name, contigs, lengths, x_contigs=[], y_contigs=[], mt_contigs=[], par=[]):
+        contigs = wrap_to_list(contigs)
+        x_contigs = wrap_to_list(x_contigs)
+        y_contigs = wrap_to_list(y_contigs)
+        mt_contigs = wrap_to_list(mt_contigs)
+        par_jrep = [interval._jrep for interval in par]
+
+        jrep = (Env.hail().variant.GenomeReference
+                .apply(name,
+                       contigs,
+                       lengths,
+                       x_contigs,
+                       y_contigs,
+                       mt_contigs,
+                       par_jrep))
+
+        self._init_from_java(jrep)
+        self._name = name
+        self._contigs = contigs
+        self._lengths = lengths
+        self._x_contigs = x_contigs
+        self._y_contigs = y_contigs
+        self._mt_contigs = mt_contigs
+        self._par = par
+
+        super(GenomeReference, self).__init__()
+
+    @handle_py4j
+    def __str__(self):
+        return self._jrep.toString()
+
+    def __repr__(self):
+        return 'GenomeReference(name=%s, contigs=%s, lengths=%s, x_contigs=%s, y_contigs=%s, mt_contigs=%s, par=%s)' % \
+               (self.name, self.contigs, self.lengths, self.x_contigs, self.y_contigs, self.mt_contigs, self.par)
+
+    @handle_py4j
+    def __eq__(self, other):
+        return self._jrep.equals(other._jrep)
+
+    @handle_py4j
+    def __hash__(self):
+        return self._jrep.hashCode()
+
+    @property
+    def name(self):
+        """Name of genome reference.
+
+        :rtype: str
+        """
+        return self._name
+
+    @property
+    def contigs(self):
+        """Contig names.
+
+        :rtype: list of str
+        """
+        return self._contigs
+
+    @property
+    def lengths(self):
+        """Dict of contig name to contig length.
+
+        :rtype: dict of str to int
+        """
+        return self._lengths
+
+    @property
+    def x_contigs(self):
+        """X contigs.
+
+        :rtype: list of str
+        """
+        return self._x_contigs
+
+    @property
+    def y_contigs(self):
+        """Y contigs.
+
+        :rtype: list of str
+        """
+        return self._y_contigs
+
+    @property
+    def mt_contigs(self):
+        """Mitochondrial contigs.
+
+        :rtype: list of str
+        """
+        return self._mt_contigs
+
+    @property
+    def par(self):
+        """Pseudoautosomal regions.
+
+        :rtype: list of :class:`.Interval`
+        """
+        return self._par
+
+    @typecheck_method(contig=strlike)
+    def contig_length(self, contig):
+        """Contig length.
+
+        :param contig: Contig
+        :type contig: str
+
+        :return: Length of contig.
+        :rtype: int
+        """
+        return self._jrep.contigLength(contig)
```

Comment:
Since you have _lengths in python already, just use that instead of calling into python.